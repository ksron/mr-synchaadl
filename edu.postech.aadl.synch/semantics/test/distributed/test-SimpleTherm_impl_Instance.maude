set show timing off .

load ../../interpreter-distributed.maude
load ../syntax/SimpleTherm_impl_Instance.maude 

---------------------------------------------
mod TEST-THERMOSTAT-SYSTEM is
  including SIMPLETHERM_IMPL_INSTANCE-MODEL .
  including MODEL-TRANSITION-SYSTEM .

  op initState : -> Object .
  eq initState = initialize(initial) .

  op cinitState : -> Object .
  eq cinitState = initialize(collapse(initial)) .

  eq #sample = 4 .
  eq #response = 4 .
endm

red initState .

red cinitState .

---set print attribute on .

rew  [100] {cinitState | 0 | infty} .

rew  [100] {cinitState | 0 | infty} .

---the number of reachable states up to 5 synchronous step
search [,5] {initState | 0 | infty} =>* {none | T:Time | infty} .

search [,5] {cinitState | 0 | infty} =>* {none | T:Time | infty} .
