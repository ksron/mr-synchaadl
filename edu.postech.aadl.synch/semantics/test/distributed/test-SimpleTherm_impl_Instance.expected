==========================================
reduce in TEST-THERMOSTAT-SYSTEM : initState .
rewrites: 374
result Object: < SimpleThermimplInstance : System | properties : ((
    TimingProperties::Period => {{5}}) ; (HybridSynchAADL::Synchronous => {{
    true}}) ; HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}),
    features : none,subcomponents : ((msg * # [false] to (environment ..
    offcontrol)) (msg * # [false] to (environment .. oncontrol)) < controller :
    System | properties : ((TimingProperties::Period => {{5}}) ; (
    HybridSynchAADL::Synchronous => {{true}}) ; (
    HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}) ; (
    HybridSynchAADL::SamplingTime => {{0.0 .. 2.0}}) ;
    HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features : (< offcontrol :
    AsyncDataOutPort | properties : none,content : * # [false],destination : (
    environment .. offcontrol) > < temp : AsyncDataInPort | properties : none,
    content : [0.0] # [false],destination : (ctrlProc .. temp),cache : [0.0] #
    [false] > < oncontrol : AsyncDataOutPort | properties : none,content : * #
    [false],destination : (environment .. oncontrol) >),subcomponents : <
    ctrlProc : Process | properties : ((TimingProperties::Period => {{5}}) ;
    HybridSynchAADL::Synchronous => {{true}}),features : (< offcontrol :
    AsyncDataOutPort | properties : none,content : * # [false],destination :
    offcontrol > < temp : AsyncDataInPort | properties : none,content : [0.0] #
    [false],destination : (ctrlThread .. temp),cache : [0.0] # [false] > <
    oncontrol : AsyncDataOutPort | properties : none,content : * # [false],
    destination : oncontrol >),subcomponents : < ctrlThread : AsyncThread |
    properties : ((TimingProperties::Period => {{5}}) ; (
    HybridSynchAADL::Synchronous => {{true}}) ; (HybridSynchAADL::SamplingTime
    => {{0.0 .. 2.0}}) ; HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),
    features : (< offcontrol : AsyncDataOutPort | properties : none,content : *
    # [false],destination : offcontrol > < temp : AsyncDataInPort | properties
    : none,content : [0.0] # [false],destination : empty,cache : [0.0] # [
    false] > < oncontrol : AsyncDataOutPort | properties : none,content : * # [
    false],destination : oncontrol >),subcomponents : (< lower : Data |
    properties : DataModel::InitialValue => {{10}},features : none,
    subcomponents : none,connections : empty,value : [1.0e+1] # [true] > <
    upper : Data | properties : DataModel::InitialValue => {{30}},features :
    none,subcomponents : none,connections : empty,value : [3.0e+1] # [true] > <
    curr : Data | properties : DataModel::InitialValue => {{15}},features :
    none,subcomponents : none,connections : empty,value : [1.5e+1] # [true] >),
    connections : empty,period : 5,perTimer : 0,msgTimer : infty,varGen : <
    "SimpleThermimplInstance.controller.ctrlProc.ctrlThread" >,currState :
    init,completeStates : init,variables : x : Real,transitions : init -[on
    dispatch]-> init {(v{x} := ((c[curr] + f[temp]) / [[2]])) ; if(v[x] < c[
    lower]){oncontrol !} (elsif(v[x] > c[upper]){offcontrol !}) end if ; c{
    curr} := v[x]},loopBound : 10,transBound : 10,execTime : 5/2,execTimer :
    infty > > > < environment : AsyncEnv | properties : ((
    TimingProperties::Period => {{5}}) ; (HybridSynchAADL::Synchronous => {{
    true}}) ; HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}),
    features : (< offcontrol : AsyncEnvInPort | properties : none,content : * #
    [false],envCache : null(Unit),destination : empty > < temp :
    AsyncEnvOutPort | properties : none,content : [0.0] # [false],envCache :
    null(Real),target : controller,destination : (controller .. temp) > <
    oncontrol : AsyncEnvInPort | properties : none,content : * # [false],
    envCache : null(Unit),destination : empty >),subcomponents : < x : Data |
    properties : DataModel::InitialValue => {{16}},features : none,
    subcomponents : none,connections : empty,value : [1.6e+1] # [true] >,
    connections : x ==> temp,period : 5,perTimer : 0,msgTimer : infty,varGen :
    < "SimpleThermimplInstance.environment" >,currMode : off,jumps : ((off -[
    offcontrol]-> off) ; (off -[oncontrol]-> on) ; (on -[offcontrol]-> off) ;
    on -[oncontrol]-> on),flows : ((off [x(t)= (c[x] - (v[t] * [[
    1.0000000000000001e-1]]))]) ; on [x(t)= (c[x] + (v[t] * [[
    2.0000000000000001e-1]]))]),sampling : controller :(0,2),response :
    controller :(3,4),samplingTimes : controller : (1/4, 3/4, 5/4, 7/4),
    responseTimes : controller : (25/8, 27/8, 29/8, 31/8),envState : idle,
    envTimer : infty >) >
==========================================
reduce in TEST-THERMOSTAT-SYSTEM : cinitState .
rewrites: 402
result Object: < SimpleThermimplInstance : System | properties : ((
    TimingProperties::Period => {{5}}) ; (HybridSynchAADL::Synchronous => {{
    true}}) ; HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}),
    features : none,subcomponents : ((msg * # [false] to (environment ..
    offcontrol)) (msg * # [false] to (environment .. oncontrol)) < environment
    : AsyncEnv | properties : ((TimingProperties::Period => {{5}}) ; (
    HybridSynchAADL::Synchronous => {{true}}) ;
    HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}),features :
    (< offcontrol : AsyncEnvInPort | properties : none,content : * # [false],
    envCache : null(Unit),destination : empty > < temp : AsyncEnvOutPort |
    properties : none,content : [0.0] # [false],envCache : null(Real),target :
    controller,destination : ((controller . ctrlProc . ctrlThread) .. temp) > <
    oncontrol : AsyncEnvInPort | properties : none,content : * # [false],
    envCache : null(Unit),destination : empty >),subcomponents : < x : Data |
    properties : DataModel::InitialValue => {{16}},features : none,
    subcomponents : none,connections : empty,value : [1.6e+1] # [true] >,
    connections : x ==> temp,period : 5,perTimer : 0,msgTimer : infty,varGen :
    < "SimpleThermimplInstance.environment" >,currMode : off,jumps : ((off -[
    offcontrol]-> off) ; (off -[oncontrol]-> on) ; (on -[offcontrol]-> off) ;
    on -[oncontrol]-> on),flows : ((off [x(t)= (c[x] - (v[t] * [[
    1.0000000000000001e-1]]))]) ; on [x(t)= (c[x] + (v[t] * [[
    2.0000000000000001e-1]]))]),sampling : controller :(0,2),response :
    controller :(3,4),samplingTimes : controller : (1/4, 3/4, 5/4, 7/4),
    responseTimes : controller : (25/8, 27/8, 29/8, 31/8),envState : idle,
    envTimer : infty > < controller . ctrlProc . ctrlThread : AsyncThread |
    properties : ((TimingProperties::Period => {{5}}) ; (
    HybridSynchAADL::Synchronous => {{true}}) ; (
    HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}) ; (
    HybridSynchAADL::SamplingTime => {{0.0 .. 2.0}}) ;
    HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features : (< offcontrol :
    AsyncDataOutPort | properties : none,content : * # [false],destination : (
    environment .. offcontrol) > < temp : AsyncDataInPort | properties : none,
    content : [0.0] # [false],destination : empty,cache : [0.0] # [false] > <
    oncontrol : AsyncDataOutPort | properties : none,content : * # [false],
    destination : (environment .. oncontrol) >),subcomponents : (< lower : Data
    | properties : DataModel::InitialValue => {{10}},features : none,
    subcomponents : none,connections : empty,value : [1.0e+1] # [true] > <
    upper : Data | properties : DataModel::InitialValue => {{30}},features :
    none,subcomponents : none,connections : empty,value : [3.0e+1] # [true] > <
    curr : Data | properties : DataModel::InitialValue => {{15}},features :
    none,subcomponents : none,connections : empty,value : [1.5e+1] # [true] >),
    connections : empty,period : 5,perTimer : 0,msgTimer : infty,varGen : <
    "SimpleThermimplInstance.controller.ctrlProc.ctrlThread" >,currState :
    init,completeStates : init,variables : x : Real,transitions : init -[on
    dispatch]-> init {(v{x} := ((c[curr] + f[temp]) / [[2]])) ; if(v[x] < c[
    lower]){oncontrol !} (elsif(v[x] > c[upper]){offcontrol !}) end if ; c{
    curr} := v[x]},loopBound : 10,transBound : 10,execTime : 5/2,execTimer :
    infty >) >
==========================================
rewrite [100] in TEST-THERMOSTAT-SYSTEM : {cinitState | 0 | infty} .
rewrites: 7481
result GlobalSystem: {< SimpleThermimplInstance : System | properties : ((
    TimingProperties::Period => {{5}}) ; (HybridSynchAADL::Synchronous => {{
    true}}) ; HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}),
    features : none,subcomponents : (< environment : AsyncEnv | properties : ((
    TimingProperties::Period => {{5}}) ; (HybridSynchAADL::Synchronous => {{
    true}}) ; HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}),
    features : (< offcontrol : AsyncEnvInPort | properties : none,content : * #
    [false],envCache : null(Unit),destination : empty > < temp :
    AsyncEnvOutPort | properties : none,content : [0.0] # [false],envCache :
    null(Real),target : controller,destination : ((controller . ctrlProc .
    ctrlThread) .. temp) > < oncontrol : AsyncEnvInPort | properties : none,
    content : * # [false],envCache : null(Unit),destination : empty >),
    subcomponents : < x : Data | properties : DataModel::InitialValue => {{
    16}},features : none,subcomponents : none,connections : empty,value : [
    1.325e+1] # [true] >,connections : x ==> temp,period : 5,perTimer : 5/2,
    msgTimer : infty,varGen : < "SimpleThermimplInstance.environment" >,
    currMode : off,jumps : ((off -[offcontrol]-> off) ; (off -[oncontrol]-> on)
    ; (on -[offcontrol]-> off) ; on -[oncontrol]-> on),flows : ((off [x(t)= (c[
    x] - (v[t] * [[1.0000000000000001e-1]]))]) ; on [x(t)= (c[x] + (v[t] * [[
    2.0000000000000001e-1]]))]),sampling : controller :(0,2),response :
    controller :(3,4),samplingTimes : controller : (1/4, 3/4, 5/4, 7/4),
    responseTimes : controller : (25/8, 27/8, 29/8, 31/8),envState : [response,
    controller],envTimer : 7/8 > < controller . ctrlProc . ctrlThread :
    AsyncThread | properties : ((TimingProperties::Period => {{5}}) ; (
    HybridSynchAADL::Synchronous => {{true}}) ; (
    HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}) ; (
    HybridSynchAADL::SamplingTime => {{0.0 .. 2.0}}) ;
    HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features : (< offcontrol :
    AsyncDataOutPort | properties : none,content : * # [false],destination : (
    environment .. offcontrol) > < temp : AsyncDataInPort | properties : none,
    content : [0.0] # [false],destination : empty,cache : [1.3475e+1] # [true]
    > < oncontrol : AsyncDataOutPort | properties : none,content : * # [false],
    destination : (environment .. oncontrol) >),subcomponents : (< lower : Data
    | properties : DataModel::InitialValue => {{10}},features : none,
    subcomponents : none,connections : empty,value : [1.0e+1] # [true] > <
    upper : Data | properties : DataModel::InitialValue => {{30}},features :
    none,subcomponents : none,connections : empty,value : [3.0e+1] # [true] > <
    curr : Data | properties : DataModel::InitialValue => {{15}},features :
    none,subcomponents : none,connections : empty,value : [
    1.3944140624999999e+1] # [true] >),connections : empty,period : 5,perTimer
    : 5/2,msgTimer : infty,varGen : <
    "SimpleThermimplInstance.controller.ctrlProc.ctrlThread" >,currState :
    init,completeStates : init,variables : x : Real,transitions : init -[on
    dispatch]-> init {(v{x} := ((c[curr] + f[temp]) / [[2]])) ; if(v[x] < c[
    lower]){oncontrol !} (elsif(v[x] > c[upper]){offcontrol !}) end if ; c{
    curr} := v[x]},loopBound : 10,transBound : 10,execTime : 5/2,execTimer :
    infty >) > | 55/2 | infty}
==========================================
rewrite [100] in TEST-THERMOSTAT-SYSTEM : {cinitState | 0 | infty} .
rewrites: 7481
result GlobalSystem: {< SimpleThermimplInstance : System | properties : ((
    TimingProperties::Period => {{5}}) ; (HybridSynchAADL::Synchronous => {{
    true}}) ; HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}),
    features : none,subcomponents : (< environment : AsyncEnv | properties : ((
    TimingProperties::Period => {{5}}) ; (HybridSynchAADL::Synchronous => {{
    true}}) ; HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}),
    features : (< offcontrol : AsyncEnvInPort | properties : none,content : * #
    [false],envCache : null(Unit),destination : empty > < temp :
    AsyncEnvOutPort | properties : none,content : [0.0] # [false],envCache :
    null(Real),target : controller,destination : ((controller . ctrlProc .
    ctrlThread) .. temp) > < oncontrol : AsyncEnvInPort | properties : none,
    content : * # [false],envCache : null(Unit),destination : empty >),
    subcomponents : < x : Data | properties : DataModel::InitialValue => {{
    16}},features : none,subcomponents : none,connections : empty,value : [
    1.325e+1] # [true] >,connections : x ==> temp,period : 5,perTimer : 5/2,
    msgTimer : infty,varGen : < "SimpleThermimplInstance.environment" >,
    currMode : off,jumps : ((off -[offcontrol]-> off) ; (off -[oncontrol]-> on)
    ; (on -[offcontrol]-> off) ; on -[oncontrol]-> on),flows : ((off [x(t)= (c[
    x] - (v[t] * [[1.0000000000000001e-1]]))]) ; on [x(t)= (c[x] + (v[t] * [[
    2.0000000000000001e-1]]))]),sampling : controller :(0,2),response :
    controller :(3,4),samplingTimes : controller : (1/4, 3/4, 5/4, 7/4),
    responseTimes : controller : (25/8, 27/8, 29/8, 31/8),envState : [response,
    controller],envTimer : 7/8 > < controller . ctrlProc . ctrlThread :
    AsyncThread | properties : ((TimingProperties::Period => {{5}}) ; (
    HybridSynchAADL::Synchronous => {{true}}) ; (
    HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}) ; (
    HybridSynchAADL::SamplingTime => {{0.0 .. 2.0}}) ;
    HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features : (< offcontrol :
    AsyncDataOutPort | properties : none,content : * # [false],destination : (
    environment .. offcontrol) > < temp : AsyncDataInPort | properties : none,
    content : [0.0] # [false],destination : empty,cache : [1.3475e+1] # [true]
    > < oncontrol : AsyncDataOutPort | properties : none,content : * # [false],
    destination : (environment .. oncontrol) >),subcomponents : (< lower : Data
    | properties : DataModel::InitialValue => {{10}},features : none,
    subcomponents : none,connections : empty,value : [1.0e+1] # [true] > <
    upper : Data | properties : DataModel::InitialValue => {{30}},features :
    none,subcomponents : none,connections : empty,value : [3.0e+1] # [true] > <
    curr : Data | properties : DataModel::InitialValue => {{15}},features :
    none,subcomponents : none,connections : empty,value : [
    1.3944140624999999e+1] # [true] >),connections : empty,period : 5,perTimer
    : 5/2,msgTimer : infty,varGen : <
    "SimpleThermimplInstance.controller.ctrlProc.ctrlThread" >,currState :
    init,completeStates : init,variables : x : Real,transitions : init -[on
    dispatch]-> init {(v{x} := ((c[curr] + f[temp]) / [[2]])) ; if(v[x] < c[
    lower]){oncontrol !} (elsif(v[x] > c[upper]){offcontrol !}) end if ; c{
    curr} := v[x]},loopBound : 10,transBound : 10,execTime : 5/2,execTimer :
    infty >) > | 55/2 | infty}
==========================================
search [, 5] in TEST-THERMOSTAT-SYSTEM : {initState | 0 | infty} =>* {none |
    T:Time | infty} .

No solution.
states: 264  rewrites: 12838
==========================================
search [, 5] in TEST-THERMOSTAT-SYSTEM : {cinitState | 0 | infty} =>* {none |
    T:Time | infty} .

No solution.
states: 264  rewrites: 10882
Bye.
